pkg/process/util/containers.go

```diff
@@ -28,6 +30,10 @@ func SetContainerSources(names []string) {
 // GetContainers autodetects the best backend from available sources
 // if the users don't specify the preferred container sources
 func GetContainers() ([]*containers.Container, error) {
+	if runtime.GOOS == "darwin" {
```

Comment:
Is this preferred over some sort of build tag?

Reply:
I tried to remove the `docker` build tag when we build the agent for macOS. 
https://github.com/DataDog/datadog-agent/pull/6225

The problem is that it makes CircleCI tests for #container-integration to fail:
https://app.circleci.com/pipelines/github/DataDog/datadog-agent/5883/workflows/4d06359e-433c-4b87-bb34-0ce70b71cffa/jobs/258336

Reply:
Lets add a note around why we're doing this.